src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -1068,153 +1095,156 @@ void func_8089CB10(EnDinofos* this, PlayState* play) {
         play->envCtx.lightSettingOverride = 11;
     }
 
-    this->actionFunc = func_8089CBEC;
+    this->actionFunc = EnDinofos_BreatheFire;
 }
 
-void func_8089CBEC(EnDinofos* this, PlayState* play) {
-    s16 temp_s3 = ((s32)Rand_CenteredFloat(0x1000) + this->actor.shape.rot.y) + this->unk_28E;
-    Vec3f sp88;
-    Vec3f sp7C;
+void EnDinofos_BreatheFire(EnDinofos* this, PlayState* play) {
+    s16 fireRotY = ((s32)Rand_CenteredFloat(0x1000) + this->actor.shape.rot.y) + this->headRotY;
+    Vec3f velocity;
+    Vec3f accel;
     ColliderJntSphElementDim* dim;
     s32 i;
     s32 end;
-    f32 temp_f20;
-    f32 temp_f22;
+    f32 sin;
+    f32 cos;
     s32 temp_s0;
 
     SkelAnime_Update(&this->skelAnime);
     Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x800);
-    temp_f20 = Math_SinS(temp_s3);
-    temp_f22 = Math_CosS(temp_s3);
+    sin = Math_SinS(fireRotY);
+    cos = Math_CosS(fireRotY);
 
-    temp_s0 = 10 - this->unk_290;
+    temp_s0 = 10 - this->attackTimer;
     temp_s0 = CLAMP_MIN(temp_s0, 0);
 
-    sp88.x = 11.0f * temp_f20;
-    sp88.y = Rand_CenteredFloat(2.0f) + -5.4f;
-    sp88.z = 11.0f * temp_f22;
-    sp7C.x = 0.9f * temp_f20;
-    sp7C.y = Rand_CenteredFloat(0.6f) + 1.4f;
-    sp7C.z = 0.9f * temp_f22;
+    velocity.x = 11.0f * sin;
+    velocity.y = Rand_CenteredFloat(2.0f) + -5.4f;
+    velocity.z = 11.0f * cos;
+    accel.x = 0.9f * sin;
+    accel.y = Rand_CenteredFloat(0.6f) + 1.4f;
+    accel.z = 0.9f * cos;
     Actor_PlaySfx_Flagged(&this->actor, NA_SE_EN_DODO_J_FIRE - SFX_FLAG);
-    EffectSsDFire_Spawn(play, &this->bodyPartsPos[DINOFOS_BODYPART_JAW], &sp88, &sp7C, 30, 22, 255 - (temp_s0 * 20), 20,
-                        3, 8);
+    EffectSsDFire_Spawn(play, &this->bodyPartsPos[DINOFOS_BODYPART_JAW], &velocity, &accel, 30, 22,
+                        255 - (temp_s0 * 20), 20, 3, 8);
 
+    // This lets the fire balls go away from Dinolfos' jaw one after the other.
     for (end = 6, i = 3; i > 0; i--) {
```

Comment:
```suggestion
    end = 6;
    for (i = 3; i > 0; i--) {
```
Does this work as well?